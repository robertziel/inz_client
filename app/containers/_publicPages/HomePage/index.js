/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';

import PropTypes from 'prop-types';

import { Container, Divider, H1, Paper } from 'components/_ui-elements';

import Search from 'components/Search';
import messages from './messages';

export default function HomePage(props) {
  const categoryId = props.match.params.id;

  return (
    <Container>
      <H1>
        <FormattedMessage {...messages.header} />
      </H1>

      <Divider />
      <Search categoryId={categoryId} />
    </Container>
  );
}

HomePage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }),
};
