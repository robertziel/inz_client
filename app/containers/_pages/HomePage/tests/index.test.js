import React from 'react';
import { IntlProvider } from 'react-intl';

import { mount } from 'enzyme';
import waitForExpect from 'wait-for-expect';

import HomePage from '../Loadable';

function mountWrapper() {
  return mount(
    <IntlProvider locale="en">
      <HomePage />
    </IntlProvider>,
  );
}

let wrapper;

beforeAll(() => {
  wrapper = mountWrapper();
});

describe('<HomePage />', () => {
  it('should render and match the snapshot', async () => {
    await waitForExpect(() => {
      wrapper.update();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
