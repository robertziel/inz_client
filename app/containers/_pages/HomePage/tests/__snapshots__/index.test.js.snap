// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `"<div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 fyVOah MuiGrid-container\\"><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 fyVOah MuiGrid-item MuiGrid-grid-xs-12\\"><h1 class=\\"H1-sc-1hnsqhn-0 bLGcwG\\"><span>This is the HomePage container!</span></h1><hr class=\\"MuiDivider-root Divider-l75xrl-0 eHOPFw\\"></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 fyVOah MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3\\"><div class=\\"MuiPaper-root MuiPaper-elevation1 Paper-t524pg-0 oIHsO MuiPaper-rounded\\"></div></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 fyVOah MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9\\"><div class=\\"MuiPaper-root MuiPaper-elevation1 Paper-t524pg-0 oIHsO MuiPaper-rounded\\"></div></div></div>"`;
